### ISSUE TITLE
Caldav: Could not open collection: 302 Moved Temporarily

### ISSUE THREAD

**Issue by freekert** [owncloud/calendar#863]
xx@home:/home/xx# cadaver
dav:!> open https://xx.nl/remote.php/caldav
WARNING: Untrusted server certificate presented for `xx.nl':
Issued to: NL
Issued by: Secure Digital Certificate Signing, StartCom Ltd., IL
Certificate is valid from Thu, 23 Apr 2015 00:02:18 GMT to Sat, 23 Apr 2016 04:09:28 GMT
Do you wish to accept the certificate? (y/n) y
Could not open collection:
302 Moved Temporarily

Whereas the same can be done for Carddav and it works fine:
dav:/remote.php/caldav/? open https://xx.nl/remote.php/carddav
WARNING: Untrusted server certificate presented for `xx.nl':
Issued to: NL
Issued by: Secure Digital Certificate Signing, StartCom Ltd., IL
Certificate is valid from Thu, 23 Apr 2015 00:02:18 GMT to Sat, 23 Apr 2016 04:09:28 GMT
Do you wish to accept the certificate? (y/n) y
Authentication required for ownCloud on server `xx.nl':
Username: xx
Password:
dav:/remote.php/carddav/> ls
Listing collection `/remote.php/carddav/': succeeded.
Coll:   addressbooks                           0  Jul 17 17:57
Coll:   principals                             0  Jul 17 17:57
dav:/remote.php/carddav/>

I found out it didn't work because Davdroid was only offering Calendar sync, not contacts sync. Recenty I did 2 things, upgrade to ownCloud 8.1 and changed some things to get an A score in ssllabs. (Hence the switch to Davdroid, the only cal/carddav client with TLS support.)

**Comment by RealRancor** [comment]
Hi,

please provide more infos about your environment like asked in https://github.com/owncloud/calendar/blob/master/CONTRIBUTING.md#short-version which is linked in the yellow box when creating a new issue.

That works on demo.owncloud.org (oC8.1 but no idea what version the calendar app is running there):

```
cadaver
dav:!> open https://demo.owncloud.org/remote.php/caldav
Authentication required for ownCloud on server `demo.owncloud.org':
Username: test
Password: test
dav:/remote.php/caldav/> ls
Listing collection `/remote.php/caldav/': succeeded.
Coll:   calendars                              0  Jul 17 21:44
Coll:   principals                             0  Jul 17 21:44
```

and also on my instance running oC 8.1.0 with contacts 0.7.2

**Comment by freekert** [comment]
Strangely, in the apps section under "Productivity" there is Calendar 0.7.2 whereas under "Activated" there is version 0.7.1.

When I deactivate 0.7.1 and activate 0.7.2 it says that the app directory already exists.

I updated to 8.1.0 following the instructions for manual update. Strange, how can there be two version, I completely remove the old 8.0.4 folder and downloaded a fresh 8.1.0.

"cat version" in /apps/calendar/appinfo gives met 0.7.1. Is 0.7.1 provided with 8.1.0? How can there be a 0.7.2 if I never updated after that? Should I manually update now? What do you recommend?

**Comment by freekert** [comment]
Ok, I have the new version now, deleting the folder manually and going to productivity a
I was able to get 0.7.2 installed automatically. The calendar works, the 302 error remains the same.

I will gather more details about my setup for you later on. Thanx for the advice this far. I'm beginnin to suspect an Nginx config error but I never changed it and got it from the ownCloud site. A 302 error does mean Nginx sends me to a wrong place, right?

**Comment by freekert** [comment]
Here are the logs of the two requests from cadaver from my first post:

root@home:/home/freek# cat /var/log/nginx/access.log | tail
127.0.0.1 - freek [18/Jul/2015:11:03:11 +0200] "OPTIONS /remote.php/carddav/ HTTP/1.1" 200 0 "-" "cadaver/0.23.3 neon/0.29.1"
127.0.0.1 - freek [18/Jul/2015:11:03:11 +0200] "PROPFIND /remote.php/carddav/ HTTP/1.1" 207 624 "-" "cadaver/0.23.3 neon/0.29.1"
127.0.0.1 - freek [18/Jul/2015:11:03:13 +0200] "PROPFIND /remote.php/carddav/ HTTP/1.1" 207 1546 "-" "cadaver/0.23.3 neon/0.29.1"
217.19.26.115 - freek [18/Jul/2015:11:03:23 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1820 "-" "Mozilla/5.0 (Macintosh) mirall/1.8.4"
217.19.26.115 - freek [18/Jul/2015:11:03:28 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 420 "-" "Mozilla/5.0 (Macintosh) mirall/1.8.4"
217.19.26.115 - freek [18/Jul/2015:11:03:31 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Macintosh) mirall/1.8.4"
127.0.0.1 - - [18/Jul/2015:11:03:46 +0200] "OPTIONS /remote.php/caldav/ HTTP/1.1" 302 5 "-" "cadaver/0.23.3 neon/0.29.1"
217.19.26.115 - freek [18/Jul/2015:11:03:53 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 1820 "-" "Mozilla/5.0 (Macintosh) mirall/1.8.4"
217.19.26.115 - freek [18/Jul/2015:11:03:58 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 420 "-" "Mozilla/5.0 (Macintosh) mirall/1.8.4"
217.19.26.115 - freek [18/Jul/2015:11:04:03 +0200] "PROPFIND /remote.php/webdav/ HTTP/1.1" 207 372 "-" "Mozilla/5.0 (Macintosh) mirall/1.8.4"

**Comment by RealRancor** [comment]
Yes, 302 is a redirect which is mostly done by your webserver. So dig into your webserver config if there are any strange rewrites / redirects done there.

**Comment by gvde** [comment]
Doesn't the 302 include a location URL to where it is redirected?

**Comment by RealRancor** [comment]
Not within the servers logfiles. But maybe cadaver can be tweaked to see the redirect location

**Comment by freekert** [comment]
I fixed it, it was not my Nginx config, I think the reason is that I neglected to disable the calendar app when doing the 8.0.4 to 8.1.0 update. After I disabled and re-enabled the app, the occ upgrade indicated something needed to be updated (it did not indicate this before.) After that it worked.

My apologies, it was just my mistake, I didn't perfectly follow the manual upgrade path. Thanx for all the answers.
